### PR TITLE
🎉 New Source - K6 Cloud [low-code cdk]

### PR DESCRIPTION
**Pull request by haithem-souala**
## What
* A new source connector K6 Cloud https://app.k6.io/

## Streams
1. Organizations
2. Tests (Metrics)

## 🚨 User Impact 🚨
N/A

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x]Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

Capture d’écran 2022-10-24 à 20 27 54

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution! Someone will review it during this week. If you have any questions please create comments so the reviewer can focus on those points.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
2 comments and ready to go! :)

File: airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    $options:
+      name: "organizations"
+      primary_key: "id"
+      path: "/v3/organizations"
+  tests_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "tests"
+      primary_key: ""
+      path: "loadtests/v2/tests"
```
[the remark this answers is not in the document]
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml]: Indeed.

File: airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/schemas/tests.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      "type": "array"
+    },
+    "k6-tests": {
+      "type": "array"
+    },
+    "load_zones": {
+      "type": "array"
+    },
+    "meta": {
+      "type": "object"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/schemas/tests.json:18]: This schema is not correct. I think users are more interested in the data inside the `k6-tests`. Even if you want to retrive all these please add the schema for k6-tests.

File: airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    retriever:
+      record_selector:
+        $ref: "*ref(definitions.tests_selector)"
+      paginator:
+        $ref: "*ref(definitions.increment_paginator)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+    $options:
+      name: "tests"
+      primary_key: "id"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml:89]: `test` schema doesn't have `id` field remove this primary key. or if you use the record inside k6-test you can keep.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments about the schema.

File: airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/schemas/tests.json (hunk 2 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/schemas/tests.json]: Fixed

File: airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml (hunk 3 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml:89]: I keep id, it is now in the schema.

File: airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      field_pointer: []
+  organizations_selector:
+    extractor:
+      field_pointer: ["organizations"]
+  projects_selector:
+    extractor:
+      field_pointer: ["projects"]
+  tests_selector:
+    extractor:
+      field_pointer: ["k6-tests"]
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml:15]: you can simplify here:
```yaml
  selector:
    extractor:
      field_pointer: 
        - "{{ options['name'] }}"
```

Then only rename the test stream.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml (hunk 4 above)
Comment (haithem-souala) [reply airbyte-integrations/connectors/source-k6-cloud/source_k6_cloud/k6_cloud.yaml]: Thanks, done!

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
Hello! Your PR is approved but didn't have the time to publish and merge it this week. As you can check in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all PRs submitted before 2-nov are eligible to win the prize. I'll be out of the office on Friday and return Monday to start publishing your contribution. Any question you can send a message in `#hacktoberfest-2022` channel in Airbyte's Slack.

Have a good weekend and thank you for this amazing contribution for Hacktoberfest 🎉

**Review by marcosmarxm — APPROVED** [review]
Thanks @haithem-souala

**Pull request merged**